### PR TITLE
Add toast message with toastify

### PR DESCRIPTION
**Pull request by RDiMag**
I think this fixed the package-lock.json issue. I had to update npm but I could not fix the package-lock file without pulling from main and starting a new branch.

### CODE REVIEW

**Review by theodorusandi — CHANGES_REQUESTED** [review]
Hmm, I think it is better to also implement the functionalities instead of just installing a library

**Comment by RDiMag** [comment]
> Hmm, I think it is better to also implement the functionalities instead of just installing a library

Do you mean including an api call for this? Janice suggested in the team chat how I can try to do this. 

Or is there something different you mean by implementing?

File: src/components/ToastMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (jpatrici) [inline src/components/ToastMessage.tsx:1]: I think you can just delete this now :)

File: src/pages/Editor/EditorPlanModal.tsx [hunk 2]
```diff
@@ -82,8 +82,14 @@ export const EditorPlanModal = ({
       await submitCallback(payload);
       fetchEditorPlans();
       handleCloseModal();
+      toast.success(`Success!`, {
```
Comment (jpatrici) [inline src/pages/Editor/EditorPlanModal.tsx:85]:
```suggestion
      toast.success(`Successfully ${type} the plan!`, {
```

File: src/pages/Editor/EditorPlanModal.tsx [hunk 3]
```diff
@@ -82,8 +82,14 @@ export const EditorPlanModal = ({
...
+        position: toast.POSITION.BOTTOM_RIGHT,
+      });
     } catch (error) {
       setErrorMessages(error as ErrorMessageInterface[]);
+      toast.error(`Error.`, {
```
Comment (jpatrici) [inline src/pages/Editor/EditorPlanModal.tsx:90]:
```suggestion
      toast.error(`Failed to ${type} the plan`, {
```

**Review by jpatrici — COMMENTED** [review]
nice if it's working! Can you also try to add it to the Make private/public
Screen Shot 2022-07-20 at 8 14 48 PM

If you go to `EditorPlans.tsx` on line 41-58, you can try adding the toast messages there too. Try to follow the message that I added below. We should be more specific in the toast message on what just got updated or failed
```
const toggleVisibility = useCallback(
    async (plan) => {
      const planStatusChange = plan.visible ? 'private' : 'public'
      try {
        await updatePlan({
          ...plan,
          visible: !plan.visible,
        });
        
        toast.success(`Plan updated to be ${planStatusChange}`, {
           position: toast.POSITION.BOTTOM_RIGHT,
        });

        fetchEditorPlans();
      } catch (error) {
        toast.error(`Failed to make plan ${planStatusChange}`, {
           position: toast.POSITION.BOTTOM_RIGHT,
        });
      }
    },
    [fetchEditorPlans]
  );
```

File: src/pages/Editor/EditorPlanModal.tsx (hunk 2 above)
Comment (jpatrici) [reply src/pages/Editor/EditorPlanModal.tsx:85]: you can add something like this on line 19
```
const toastMsgMap = {
     add: {
        success: 'Successfully added the plan!'
        error: 'Failed to add plan'
     },
     delete: {
        success: 'Successfully deleted the plan!'
        error: 'Failed to delete plan'
     },
     update: {
        success: 'Successfully updated the plan!'
        error: 'Failed to update plan'
     }
}
```
and then do this for the toast
```
toast.success(toastMsgMap[type].success, {
   position: toast.POSITION.BOTTOM_RIGHT,
});

toast.error(toastMsgMap[type].error, {
   position: toast.POSITION.BOTTOM_RIGHT,
});
```

**Comment by Marcellia** [comment]
Nice, I like it. Do add some screenshots once you update the requested changes.  Implementing it without a library will be a good learning experience. However, I don’t mind the library as it accounts for accessibility, and lot more functionalities that we might have not considered.

**Review by Marcellia — COMMENTED** [review]
Nice, I like it. Do add some screenshots once you update the requested changes. Implementing it without a library will be a good learning experience. However, I don’t mind the library as it accounts for accessibility, and lot more functionalities that we might have not considered.

File: src/pages/Editor/EditorPlanModal.tsx (hunk 3 above)
Comment (RDiMag) [reply src/pages/Editor/EditorPlanModal.tsx:90]: Updated!

**Comment by RDiMag** [comment]
toastify-sucess

Here is how the success message looks!

**Comment by RDiMag** [comment]
error-toast

**Review by Marcellia — APPROVED** [review]

File: src/components/ToastMessage.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+function ToastMessage() {
+  const notify = () => toast('Error');
+
+  return (
+    <div>
+      <button onClick={notify}>Error</button>
+      <ToastContainer />
+    </div>
+  );
+}
```
Comment (jpatrici) [inline src/components/ToastMessage.tsx:13]: can you delete this since it's not used?

**Review by jpatrici — APPROVED** [review]

File: src/components/ToastMessage.tsx (hunk 4 above)
Comment (RDiMag) [reply src/components/ToastMessage.tsx]: Done!

**Review by theodorusandi — APPROVED** [review]
the code looks good to me

**Pull request merged**